test/test_torch.py

```diff
@@ -141,9 +133,41 @@ def test_ot_solve():
             b = ot.torch.unif(n, dtype=dtype, device=device)
 
             for metric in lst_metrics:
-
                 M = ot.torch.dist(x, y, metric)
                 G = ot.torch.ot_solve(a, b, M)
 
                 np.testing.assert_allclose(ot.unif(n), G.sum(1).cpu().numpy())
                 np.testing.assert_allclose(ot.unif(n), G.sum(0).cpu().numpy())  # cf convergence sinkhorn
+
+
+@pytest.mark.skipif(nogo, reason="Missing pytorch")
```

Comment:
this test is great but it should also check that the loss can be used with backward and that gradients for w an y are available

Reply:
I don't see how this could not but sure. I have some code that computes the gradient manually, do you think that's an overkill for a test and I should just compare with the numerical one?

```python
@nb.jit
def emd_1d(w_x, w_y, x, y):
    """EMD 1D Specialized to square euclidean distance with manual forward mode gradient"""
    n = w_x.shape[0]
    m = w_y.shape[0]
    
    idx_x = np.argsort(x)
    idx_y = np.argsort(y)
    
    sorted_x = x[idx_x]
    sorted_y = y[idx_y]
    
    sorted_w_x = w_x[idx_x]
    sorted_w_y = w_y[idx_y]
    
    cost = 0.
    grad = np.zeros(n)

    i = j = 0
    w_i = sorted_w_x[0]
    w_j = sorted_w_y[0]
    
    m_ij = 0.
    while True:
        diff = sorted_x[i] - sorted_y[j]
        m_ij = (diff) ** 2
        
        if w_i < w_j or j == m - 1:
            cost += m_ij * w_i
            grad[idx_x[i]] += 2 * w_i * diff
            i += 1
            if i == n:
                break
            w_j -= w_i
            w_i = sorted_w_x[i]
        else:
            cost += m_ij * w_j
            grad[idx_x[i]] += 2 * w_j * diff
            j += 1
            if j == m:
                break
            w_i -= w_j
            w_j = sorted_w_y[j]

    return cost, grad

```

Reply:
this is  abit overkill, maybe you should just compute them et test that with a small step the loss decreases

Reply:
You think? I was thinking either my method or this utility testing function:
https://github.com/pytorch/pytorch/blob/master/torch/autograd/gradcheck.py

Reply:
plus also simple thing like the gradient of the OT loss between two equal distributions is 0

Reply:
very good idea

I shoudl do that for my functions and i will